### PR TITLE
Quarantined device debugging

### PR DESCRIPTION
**Pull request by ricardoamador**
## Description

Adding debugging within device doctor to see if we are or not finding a device. The swarming bot is seeing success when the device_doctor recovery is run even though the device is shown as quarantined.

So far I have only observed this with mac io devices.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: device_doctor/lib/src/ios_device.dart [hunk 1]
```diff
@@ -41,13 +41,16 @@ class IosDeviceDiscovery implements DeviceDiscovery {
 
   @override
   Future<List<Device>> discoverDevices({Duration retryDuration = const Duration(seconds: 10)}) async {
-    return LineSplitter.split(await deviceListOutput()).map((String id) => IosDevice(deviceId: id)).toList();
+    List<Device> discoveredDevices = LineSplitter.split(await deviceListOutput()).map((String id) => IosDevice(deviceId: id)).toList();
+    stdout.write("ios devices discovered: $discoveredDevices");
```
Comment (keyonghan) [inline device_doctor/lib/src/ios_device.dart:46]:
```suggestion
    stdout.write("ios devices discovered: ${discoveredDevices.map(e => e.deviceId).tolist()}");
```

File: device_doctor/lib/src/ios_device.dart [hunk 2]
```diff
@@ -41,13 +41,16 @@ class IosDeviceDiscovery implements DeviceDiscovery {
 
   @override
   Future<List<Device>> discoverDevices({Duration retryDuration = const Duration(seconds: 10)}) async {
-    return LineSplitter.split(await deviceListOutput()).map((String id) => IosDevice(deviceId: id)).toList();
+    List<Device> discoveredDevices = LineSplitter.split(await deviceListOutput()).map((String id) => IosDevice(deviceId: id)).toList();
```
Comment (keyonghan) [inline device_doctor/lib/src/ios_device.dart]: final

**Review by keyonghan — COMMENTED** [review]
Chromium config also needs to pick up the `stdout`: https://chrome-internal.googlesource.com/infradata/config/+/refs/heads/main/configs/chromium-swarm/scripts/flutter_ios.py#92

**Comment by ricardoamador** [comment]
Hmmm okay I was more interested in recover but I will add the stdout to the get_state call. Thanks!

**Comment by ricardoamador** [comment]
Yeah I can wait to land.

**Pull request merged**